flow/layers/opacity_layer.cc

```diff
@@ -39,7 +39,12 @@ void OpacityLayer::Paint(PaintContext& context) const {
       RasterCache::GetIntegralTransCTM(context.canvas->getTotalMatrix()));
 #endif
 
-  if (layers().size() == 1 && context.raster_cache) {
+  // Embedded platform views are changing the canvas in the middle of the paint
+  // traversal. To make sure we paint on the right canvas, when the embedded
+  // platform views preview is enabled (context.view_embedded is not null) we
+  // don't use the cache.
+  if (context.view_embedder == nullptr && layers().size() == 1 &&
```

Comment:
Yes, this ensures that `Paint` will be called for every layer. However, I would encourage to disable the cache in Preroll instead of Paint. That way, we don't spend time on generating the cache that we're never going to use.

Reply:
Thanks good point!
My plan here is to eventually only disable it if there's a platform view in the scene (which we can ask the view_embedder only after Preroll), as a safe first step I just always disable it...